### PR TITLE
Proj1200: Exclude private assets as project file dependency

### PR DESCRIPTION
**Pull request by Corniel**
Documentation for https://github.com/dotnet-project-file-analyzers/dotnet-project-file-analyzers/pull/88

### CODE REVIEW

File: rules/Proj1200.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Proj1200: Exclude private assets as project file dependency
+Private assets (such as analyzers) are package references that have an affect
```
Comment (CptWesley) [inline rules/Proj1200.md:2]: "that affect" or "that have an effect"

File: rules/Proj1200.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+on compiling a .NET assembly, but are not needed by the compiled artifact. By
+marking them in the project file as private asset it will not become a runtime
+dependency, that linked to a package, or shiped when publishing it.
```
Comment (CptWesley) [inline rules/Proj1200.md:5]: "shipped"

**Comment by CptWesley** [comment]
LGTM if changes are applied

File: rules/Proj1200.md (hunk 2 above)
Comment (LauraXiulan) [inline rules/Proj1200.md:5]: ... that **is** linked to a package ...

File: rules/Proj1200.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+## Non-compliant
+``` XML
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PackageReference Include="DotNetProjectFile.Analyzers" Version="*" />
+
+</Project>
+```
+
+## Compliant
```
Comment (LauraXiulan) [inline rules/Proj1200.md:16]: What about IncludeAssets and ExcludeAssets? And what about other values apart from "all" ("none", "contentfiles", "analyzers", "build")

Comment (Corniel) [reply rules/Proj1200.md:16]: Theoretically you could have other values, but this rule says that you should use "all".

**Review by CptWesley — APPROVED** [review]

**Pull request merged**